include/mata/parser/mintermization.hh

```diff
@@ -17,9 +17,9 @@
 #ifndef MATA_MINTERM_HH
```

Comment:
I was wondering about this as well. I am fine either way. What do others think? I decided to put it into the `parser` directory because it works on the intermediate automaton and in theory it is a part of the parsing process due to the mintermization being basically a preprocessing part of the automata creation right after parsing… So, is it still part of the parsing, or not… I do not know. @tfiedor Do you have an idea for how to name the folder for mintermization? Or we could rename the folder `parser` into something more general which would encapsulate parser, intermediate automaton and mintermization.

Reply:
No, I would keep `parser` as parser, or as we discussed in past as some kind of `factory`, which will encapsulate functions that "build" automata (along with building sigma star, one letter automata, etc.). I would not even say that mintermization is preprocessing, though it might be closest (I mean I think you can perform it anytime). Also it is rather universal (i.e. both for NFA and AFA), but I would not add it to utils. So maybe in the end `preprocessing` or `transformers` or `canonical` (since it is related to forms)?

I think @martinhruska and @jurajsic could have some ideas.

Reply:
Or I can simply create a folder for intermediate automaton and operations on it (`intermediate-aut`). That would solve the issue, would it not?

Reply:
Resolved in our meetings:
- Mintermization and intermediate automaton will remain in the parser for now. They will be moved to a better folder when they are being rewritten.
- Builder for NFAs was added as a namespace within `Mata::Nfa` in its own file.